File: code/java/ch15/practice/Practice01ContainerMostWater.java
package ch15.practice;

import java.util.*;

/**
 * Practice 1: Container With Most Water
 * Chapter 15: Two Pointers & Sliding Window — The Caterpillar Method
 *
 * PROBLEM: Find two lines forming a container holding the most water.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Practice01ContainerMostWater {
    public static int solve(int[] heights) {
        // TODO: Replace this with your solution
        return 0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine().trim();
        int[] arr = Arrays.stream(line.split(" ")).mapToInt(Integer::parseInt).toArray();
        System.out.println(solve(arr));
        sc.close();
    }
}

File: code/python/ch15/practice/practice_01_container_most_water.py
"""
Practice 1: Container With Most Water
=======================================
Chapter 15: Two Pointers & Sliding Window — The Dance of Indices

PROBLEM
-------
Given an array of non-negative integers where each element represents
the height of a vertical line, find two lines that together with the
x-axis form a container holding the most water. Return the maximum area.

INPUT FORMAT
------------
A single line of space-separated non-negative integers.

OUTPUT FORMAT
-------------
A single integer — the maximum water area.

CONSTRAINTS
-----------
- 2 <= len(heights) <= 10^5
- 0 <= heights[i] <= 10^4

EXAMPLES
--------
Input:
  1 8 6 2 5 4 8 3 7
Output: 49

Input:
  1 1
Output: 1

INSTRUCTIONS
------------
Replace the `pass` in the solve() function with your solution.
The main block at the bottom handles input/output — don't change it.
"""


def solve(heights: list[int]) -> int:
    """Return maximum water area."""
    pass  # TODO: Replace this with your solution



# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    line = input().strip()
    arr = list(map(int, line.split()))
    print(solve(arr))

